test/gateways/trexle_test.exs

```diff
@@ -87,6 +358,16 @@ defmodule Gringotts.Gateways.TrexleTest do
   end
 
   describe "authorize" do
+    test "with unauthorized access." do
```

Comment:
Why was this written using `Mock`?

Reply:
This was written using mock because this test case just tests our code's  functionality when 401 unauthorised is returned. In my opinion, there is no need here to use bypass because request parameters are already tested above using bypass. Bypass creates a local server in the background and then test request parameters. So to avoid all that processing I used mock here.
Let me know your thoughts.

Reply:
Since, mock is slower than bypass. Done